types/d3-array/index.d.ts

```diff
@@ -58,12 +64,12 @@ export function min<T extends Numeric>(array: ArrayLike<T>): T | undefined;
 /**
  * Return the minimum value in the array using natural order.
  */
-export function min<T>(array: ArrayLike<T>, accessor: (datum: T, index: number, array: ArrayLike<T>) => string | undefined | null): string | undefined;
+export function min<K extends AnArray<any>>(array: K, accessor: (datum: K[0], index: number, array: K) => string | undefined | null): string | undefined;
```

Comment:
I guess we can even go back down to:
```ts
export function max<T>(array: ArrayLike<T>, accessor: (datum: T, index: number, array: ArrayLike<T>) => string | undefined | null): string | undefined;
export function max<T, U extends Numeric>(array: ArrayLike<T>, accessor: (datum: T, index: number, array: ArrayLike<T>) => U | undefined | null): U | undefined;
```
And simply fix the accessor functions in the test file to use `ArrayLike<MixedObject>` rather than `MixedObject[]`